packages/engine.io-client/lib/socket.ts

```diff
@@ -717,6 +750,9 @@ export class SocketWithoutUpgrade extends Emitter<
    * @return {Socket} for chaining.
    */
   public write(msg: RawData, options?: WriteOptions, fn?: () => void) {
+    // this will schedule a disconnection on next tick if heartbeat missed
+    this.isResponsive();
```

Comment:
Is this really needed? The write will fail anyway, no?

Reply:
The problem is without this the write will fail but then the packet is lost and won’t be retried

With this the packet gets queued and sent when the connection is restored